### PR TITLE
On finalize session

### PR DESCRIPTION
**Pull request by korzewski**

### CODE REVIEW

File: pallets/jackblock/src/lib.rs [hunk 1]
```diff
@@ -81,4 +106,76 @@ decl_module! {
...
+		debug::info!("--- winners: {:?}", winners);
+
+		Ok(())
+	}
+
+	fn get_session_numbers() -> GuessNumbersType {
+		let mut session_numbers: GuessNumbersType = [0; GUESS_NUMBERS_COUNT];
+		for i in 0..6 {
+			session_numbers[i] = Self::get_random_number(i as u8);
+		}
```
Comment (tk-o) [inline pallets/jackblock/src/lib.rs:130]:
```suggestion
      let mut i = 0;
		loop {
			let next_session_number = Self::get_random_number(i as u8);
			// only add number to session numbers if it doesn't exist yet
			if !session_numbers.includes(&next_session_number) {
			    session_numbers[i] = next_session_number;
			    i = i + 1;
			}
			
			if i == GUESS_NUMBERS_COUNT {
			    break;
			}
		}
```

File: pallets/jackblock/src/lib.rs [hunk 2]
```diff
@@ -81,4 +106,76 @@ decl_module! {
...
+					correct += 1;
+				}
+			}
+			
+			if correct > 0 {
+				winners.push((bet, correct));
+			}
+		}
+
+		winners
```
Comment (tk-o) [inline pallets/jackblock/src/lib.rs:152]: Try using iter + filter + map + collect to make the code more declarative

Comment (korzewski) [reply pallets/jackblock/src/lib.rs]: good point, thanks!

**Review by tk-o — APPROVED** [review]

**Review by deployAt — APPROVED** [review]

File: pallets/jackblock/src/lib.rs [hunk 3]
```diff
@@ -146,19 +146,16 @@ impl<T: Config> Module<T> {
...
-			if correct > 0 {
-				winners.push((bet, correct));
-			}
-		}
-
-		winners
+				(bet, correct)
+			})
+			.filter(|x| x.1 > 0)
+			.collect::<Winners<T::AccountId>>()
```
Comment (deployAt) [inline pallets/jackblock/src/lib.rs:158]: me gusta

**Pull request merged**